### PR TITLE
Enable users to sign multisignature transactions - closes #3092

### LINKED ISSUE #3092: Enable users to sign multisignature transactions
### Description
Users must be able to sign transactions of multisignature groups.

[UI Design](https://www.figma.com/proto/kNu67yxyOZCJKU21fgFvBl/Multisignature?node-id=228%3A21963&viewport=145%2C516%2C0.05319149047136307&scaling=min-zoom)

### Acceptance Criteria
- There should be a new button on the side bar called `Sign a transaction` to open the `Sign multisignature transaction modal`.
- The modal should enable users to
   - Paste the content of a multisignature transaction.
   - Read the content of a multisignature transaction as a JSON file from their computer.
   - Review the details of the transaction.
   - Sign the transaction.
   - Copy the signed transaction content.
   - Download the signed transaction content as a JSON file.

### PR DESCRIPTION
**Pull request by isalga**
### What was the problem?
This PR resolves #3092 

### How was it solved?
- Add modal
- Add sidebar menu
- Add UI components
- Validate JSON input
- Sign transaction using lisk-core
- Copy and download JSON transaction

### How was it tested?
- Add unit test

### CODE REVIEW

File: src/components/screens/signMultiSignTransaction/importData/importData.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import BoxFooter from '../../../toolbox/box/footer';
+import { PrimaryButton } from '../../../toolbox/buttons';
+import ProgressBar from '../progressBar';
+import styles from './styles.css';
+
+const ImportData = ({ t, nextStep }) => {
+  const [jsonInput, setJsonInput] = useState(undefined);
+  const reader = new FileReader();
+
+  reader.onload = ({ target }) => {
```
Comment (UsamaHameed) [inline src/components/screens/signMultiSignTransaction/importData/importData.js:13]: I think we should move `const reader = new FileReader();` to out of the component and add listeners to the reader instance in `useEffect`

File: src/components/screens/signMultiSignTransaction/importData/importData.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    const parsedInput = JSON.parse(target.result);
```
Comment (UsamaHameed) [inline src/components/screens/signMultiSignTransaction/importData/importData.js:14]: Should we validate the json the user puts here?

File: src/components/screens/signMultiSignTransaction/reviewSign/reviewSign.test.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  });
+
+  it('Should call props.prevStep', () => {
+    wrapper.find('button.go-back').simulate('click');
+    expect(props.prevStep).toBeCalled();
+  });
+
+  it('Should render properly', () => {
+    const html = wrapper.html();
+    expect(wrapper.find('.member-info').length).toEqual(props.members.length);
```
Comment (UsamaHameed) [inline src/components/screens/signMultiSignTransaction/reviewSign/reviewSign.test.js:41]:
```suggestion
	expect(wrapper).toContainMatchingElements(props.members.length, '.member-info');
```

**Review by UsamaHameed — CHANGES_REQUESTED** [review]

**Comment by UsamaHameed** [comment]
Do we not also need to add an icon to the side bar to activate this modal from?

**Review by UsamaHameed — DISMISSED** [review]

**Review by reyraa — CHANGES_REQUESTED** [review]
- It doesn't allow pasting values. I can't click on the input area and paste my tx value.
- It doesn't read a valid tx JSON file. I have the following content (reading from `tx.json`) which is a valid unsigned tx and I should be able to sign it using this tool, but it shows "Invalid file" error:
```
{"nonce":"157","fee":"10000000","type":8,"asset":{"amount":"100000000","recipientId":"5059876081639179984L"}}
```

File: src/components/screens/signMultiSignTransaction/importData/importData.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    optionalKeys, numberOfSignatures,
+  } = asset;
+
+  if (
+    parseInt(nonce, 10) && parseInt(fee, 10)
+    && type === transactionTypes().transfer.code.new && parseInt(amount, 10)
+    && Array.isArray(signatures) && Array.isArray(optionalKeys) && Array.isArray(mandatoryKeys)
+    && typeof recipientId === 'string' && typeof data === 'string'
+    && typeof lsTrackingId === 'string' && typeof senderPublicKey === 'string'
+    && typeof numberOfSignatures === 'number'
```
Comment (reyraa) [inline src/components/screens/signMultiSignTransaction/importData/importData.js:28]: I would go with a list of validators and `Array.reduce`.  It's more readable, extendible and definitely easier tested.

File: src/components/screens/signMultiSignTransaction/importData/index.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import ImportDataComp from './importData';
+
+const ImportData = (props) => {
+  const { t } = useTranslation();
+  return <ImportDataComp t={t} {...props} />;
+};
```
Comment (reyraa) [inline src/components/screens/signMultiSignTransaction/importData/index.js:8]: Why not simply wrap in `withTranslate`?

File: src/components/screens/signMultiSignTransaction/reviewSign/index.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import ReviewSignComp from './reviewSign';
+
+const ReviewSign = (props) => {
+  const { t } = useTranslation();
+  return <ReviewSignComp t={t} {...props} />;
```
Comment (reyraa) [inline src/components/screens/signMultiSignTransaction/reviewSign/index.js:7]: Same question, why not `withTranslate`?

File: src/components/screens/signMultiSignTransaction/importData/importData.js [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+const reader = new FileReader();
+
+// eslint-disable-next-line max-statements
+const inputValidator = ({
+  senderId, senderPublicKey, fee, asset,
+  nonce, signatures, lsTrackingId, id,
+}) => {
+  const validation = {
+    valid: true,
```
Comment (UsamaHameed) [inline src/components/screens/signMultiSignTransaction/importData/importData.js:19]: Do we really need a boolean here? Can we not just directly check `errors.length`?

**Review by UsamaHameed — CHANGES_REQUESTED** [review]

File: src/components/screens/signMultiSignTransaction/importData/importData.js (hunk 7 above)
Comment (isalga) [reply src/components/screens/signMultiSignTransaction/importData/importData.js:19]: In the future we will have to replace this validation for `Lisk.utils.validateMultisignatures` which returns an object like this. This is why I decided to keep that structure.

**Review by UsamaHameed — APPROVED** [review]

**Review by reyraa — APPROVED** [review]

**Pull request merged**